ocaml-lsp-server/src/merlin_query.ml

```diff
@@ -0,0 +1,21 @@
+open Fiber.O
+
+let doc_comment doc pos =
+  let+ res =
+    let command = Query_protocol.Document (None, pos) in
+    Document.dispatch_exn doc command
+  in
+  match res with
+  | `Found s
+  | `Builtin s ->
+    Some s
+  | _ -> None
+
+let type_enclosing doc pos =
+  let command = Query_protocol.Type_enclosing (None, pos, None) in
+  let+ res = Document.dispatch_exn doc command in
+  match res with
+  | []
+  | (_, `Index _, _) :: _ ->
+    None
+  | (location, `String value, _) :: _ -> Some (location, value)
```

Comment:
perhaps move this to Document? This module doesn't really have a well defined purpose on its own.